File: Data Structures & Algorithms/subarray-sum-equals-k/submission-6.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = defaultdict(int)
        res = curSum = 0
        for x in nums:
            count[curSum] += 1
            curSum += x
            res += count[curSum - k]
        return res

File: Data Structures & Algorithms/concatenation-of-array/submission-6.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] res = new int[2 * nums.length];
        int idx = 0;
        for(int i = 0; i < 2; i++){
            for(int n: nums){
                res[idx++] = n;
            }
        }
        return res;
    }
}